### DISCUSSION TITLE
Zoom to Element does not highlight sufficiently to identify some elements

### DISCUSSION THREAD

**Discussion by candcconsulting** [iTwin/itwinjs-core#1522] [Q&A] [answered]
We have some code that identifies assets in the iModel and then collects the elements and zooms to those elements.

For some assets it is extremely difficult to identify the elements.  This is the code we are using to zoom to and emphasize the elements ...
```
    onMouseButton(_ev: BeButtonEvent): boolean {
        if (!_ev.isDown) return true;
        const emph = EmphasizeElements.getOrCreate(IModelApp.viewManager.selectedView!);        
        IModelApp.viewManager.selectedView!.zoomToElements(this._elementId, { animateFrustumChange: true, standardViewId: StandardViewId.RightIso});
        emph.emphasizeElements(this._elementId, IModelApp.viewManager.selectedView!, undefined, true)
//        IModelConnection.selectionSet.replace(this._elementId)
        return true;
    }
```
However, we notice that "selecting" the elements highlights the elements much better, but this seems to be through IModelConnection rather than IModelApp ... so having to pass the connection into the class
Is there some way to set emphasizeElements so it can be similar to selecting the emphasized elements or some way to get the IModelConnection from the IModelApp
27s onwards in the video shows the problem

https://user-images.githubusercontent.com/19268085/119821307-50abe780-beea-11eb-890f-47f610b9482e.mp4

**Comment by pmconne** [comment]
I am probably dense, but I don't fully understand the problem.
Is it that the settings used for emphasize (default: thick black outline, no color change) are harder to see than those used for hilite (default: thin blue outline, blue tint)?
If so, you can [customize those settings](https://www.itwinjs.org/reference/imodeljs-frontend/views/viewport/emphasissettings/) to your liking.
If not, please clarify the problem.

**Reply by candcconsulting** [reply]
Not sure you are dense .,.. did you mean something like this ?
```
    onMouseButton(_ev: BeButtonEvent): boolean {
        if (!_ev.isDown) return true;
        let emphasis = new Hilite.Settings(ColorDef.blue, 1, 1, Hilite.Silhouette.Thick);        
        IModelApp.viewManager.selectedView!.emphasisSettings = emphasis;
        const emph = EmphasizeElements.getOrCreate(IModelApp.viewManager.selectedView!);        
        IModelApp.viewManager.selectedView!.zoomToElements(this._elementId, { animateFrustumChange: true, standardViewId: StandardViewId.RightIso});
        emph.emphasizeElements(this._elementId, IModelApp.viewManager.selectedView!, undefined , true)
//        IModelConnection.selectionSet.replace(this._elementId)
        return true;
    }
```
This does not seem to change the emphasis of the elements ...

**Reply by pmconne** [reply]
@candcconsulting see @bbastings' reply below...by default `EmphasizeElements` does not apply the outlined "emphasis" effect - you need to enable it.

**Comment by bbastings** [comment, marked as answer]
Sounds like you might want to enable wantEmphasise for EmphasizeElements to further "hilite" the elements:

  /** If true, all overridden and emphasized elements will also have the "emphasis" effect applied to them. This causes them to be hilited using the current [[Viewport.emphasisSettings]]. */
  public wantEmphasis = false;

**Reply by candcconsulting** [reply]
Sorry bbastings how is that set ... If I try and add a featureAppearance
`let featureApp = FeatureAppearance.fromJSON({emphasized: true, rgb: {r: 255, g: 0 , b:0}, transparency: 0, weight: 15})
`then this just turns everything blue, and it will not let me set wantEmphasis on a viewPort ...

**Reply by pmconne** [reply]
See added line below.
```
    onMouseButton(_ev: BeButtonEvent): boolean {
        if (!_ev.isDown) return true;
        const emph = EmphasizeElements.getOrCreate(IModelApp.viewManager.selectedView!);

/*=>*/  emph.wantEmphasis = true;

        IModelApp.viewManager.selectedView!.zoomToElements(this._elementId, { animateFrustumChange: true, standardViewId: StandardViewId.RightIso});
        emph.emphasizeElements(this._elementId, IModelApp.viewManager.selectedView!, undefined, true)

        return true;
    }
```

**Reply by candcconsulting** [reply]
Thanks that puts a highlight perimeter around the elements which really helps

**Comment by candcconsulting** [comment]
turn on wantEmphasis
https://github.com/imodeljs/imodeljs/discussions/1522#discussioncomment-795195
